packages/flutter/test/material/snack_bar_theme_test.dart

```diff
@@ -286,6 +286,136 @@ void main() {
     expect(snackBarBottomRight.dx, (800 + snackBarWidth) / 2); // Device width is 800.
   });
 
+  testWidgets('Local SnackBarTheme properties are used', (WidgetTester tester) async {
+    const Color backgroundColor = Colors.purple;
+    const Color textColor = Colors.pink;
+    const elevation = 7.0;
+    const action = 'ACTION';
+    const ShapeBorder shape = RoundedRectangleBorder(
+      borderRadius: BorderRadius.all(Radius.circular(9.0)),
+    );
+    const snackBarWidth = 400.0;
+
+    await tester.pumpWidget(
+      MaterialApp(
+        home: SnackBarTheme(
```

Comment:
It seems it only works when `SnackBarTheme` wraps `Scaffold`. What if it's inside Scaffold? I think it's a bit rare to use SnackBarTheme to wrap a Scaffold like this. If that's the only use case, then I don't think it's necessary to add this class. We can directly use MaterialApp.theme.